pandas/tests/frame/indexing/test_indexing.py

```diff
@@ -1737,6 +1737,13 @@ def test_getitem_preserve_object_index_with_dates(self, indexer):
 
         assert ser.index.dtype == object
 
+    def test_loc_setitem_empty_frame_expansion(self):
+        # GH#49972
+        result = DataFrame()
+        result.loc[0, 0] = np.asarray([0])
+        expected = DataFrame({0: [0.0]})
```

Comment:
Hm in principal I agree, but this is consistent now with
```
df = DataFrame({0: [0]})
df.loc[1, 0] = np.asarray([0])
```

and 

```
df = DataFrame({0: [0]})
df.loc[0, 1] = np.asarray([0])
```

which are similar expanding cases. This points to the bug being somewhere else regarding scalar vs array

Reply:
thats reasonable.  maybe let's see what it would take to fix the two cases you mention here and whether those are isolate-ish sketchy cases before adding another one?

if moving forward with this, should add a comment in test/code about living with technically-wrong behavior

Reply:
There is a general inconsistency that might be expected? Not Sure.

If you have a column with a numeric dtype

```
df = DataFrame({0: [0]})
```

then

```
df.loc[1, 0] = np.asarray([0])
```
this unpacks and sets a scalar. If the column has object dtype,

```
df = DataFrame({0: ["a"]})
```

we set the array. Is this expected?

Reply:
> Is this expected?

uhh maybe?  with object-dtype it is more plausible that the user is intentionally trying to set an array as a value.